pkg/steps/cluster_claim.go

```diff
@@ -212,31 +213,60 @@ func (s *clusterClaimStep) releaseCluster(ctx context.Context, clusterClaim *hiv
 	return nil
 }
 
-func (s *clusterClaimStep) saveArtifacts(ctx context.Context, namespace, name string) error {
+func (s *clusterClaimStep) saveArtifacts(ctx context.Context, namespace, name string, printConditions bool) error {
 	logrus.WithField("clusterClaim.Namespace", namespace).WithField("clusterClaim.Name", name).Debug("Saving artifacts.")
 	var errs []error
 	namespaceDir := api.NamespaceDir
-	claim := &hivev1.ClusterClaim{}
+	claim := hivev1.ClusterClaim{}
 	if err := s.saveObjectAsArtifact(
 		ctx,
 		ctrlruntimeclient.ObjectKey{Namespace: namespace, Name: name},
-		claim,
+		&claim,
 		"cluster claim",
 		filepath.Join(namespaceDir, "clusterClaim.json"),
 	); err != nil {
 		errs = append(errs, err)
 	}
+	if printConditions {
+		builder := &strings.Builder{}
+		_, _ = builder.WriteString(fmt.Sprintf("Found %d conditions for ClusterClaim:", len(claim.Status.Conditions)))
+		sort.Slice(claim.Status.Conditions, func(i, j int) bool {
+			return claim.Status.Conditions[i].LastTransitionTime.Before(&claim.Status.Conditions[j].LastTransitionTime)
+		})
+		for _, condition := range claim.Status.Conditions {
+			if condition.Status == corev1.ConditionUnknown || condition.Reason == "Initialized" {
+				continue
+			}
+			_, _ = builder.WriteString(fmt.Sprintf("\n[%s]%s: %s", condition.LastTransitionTime.Format(time.RFC3339), condition.Reason, condition.Message))
+		}
+		logrus.Warn(builder.String())
+	}
 
 	if claim.Spec.Namespace != "" {
+		clusterDeployment := hivev1.ClusterDeployment{}
 		if err := s.saveObjectAsArtifact(
 			ctx,
 			ctrlruntimeclient.ObjectKey{Namespace: claim.Spec.Namespace, Name: claim.Spec.Namespace},
-			&hivev1.ClusterDeployment{},
+			&clusterDeployment,
 			"cluster deployment",
 			filepath.Join(namespaceDir, "clusterDeployment.json"),
 		); err != nil {
 			errs = append(errs, err)
 		}
+		if printConditions {
+			builder := &strings.Builder{}
+			_, _ = builder.WriteString(fmt.Sprintf("Found %d conditions for ClusterDeployment:", len(clusterDeployment.Status.Conditions)))
+			sort.Slice(clusterDeployment.Status.Conditions, func(i, j int) bool {
```

Comment:
NIT: SliceStable. Not sure if the origin order (when 2 conditions have the same transition time) has some meaning though.
```suggestion
			sort.SliceStable(clusterDeployment.Status.Conditions, func(i, j int) bool {
```

Reply:
We're just printing, it does not really matter that this is stable